src/plugins/data/public/search/expressions/esaggs.ts

```diff
@@ -302,18 +302,15 @@ export const esaggs = (): EsaggsExpressionFunctionDefinition => ({
       abortSignal: (abortSignal as unknown) as AbortSignal,
     });
 
-    const table: KibanaDatatable = {
-      type: 'kibana_datatable',
+    const table: Datatable = {
+      type: 'datatable',
       rows: response.rows,
       columns: response.columns.map((column: any) => {
-        const cleanedColumn: KibanaDatatableColumn = {
+        const cleanedColumn: DatatableColumn = {
           id: column.id,
           name: column.name,
           meta: serializeAggConfig(column.aggConfig),
         };
-        if (args.includeFormatHints) {
-          cleanedColumn.formatHint = column.aggConfig.toSerializedFieldFormat();
-        }
```

Comment:
This property is important to us in Lens. I haven't run the code for this PR yet, but I think this would break our formatting.

Reply:
That’s a great point. @ppisljar I think we didn’t really consider this in the discussion of the data table format. If you are hesitant to make the format hint a top level property of data table columns, we could also make it a part of the data source specific “meta” property.

Reply:
ATM I don't think there are any plans to merge this PR as-is. I believe Peter is reworking the structure to this a bit as we found a few holes with the plan in https://github.com/elastic/kibana/issues/68457.

> If you are hesitant to make the format hint a top level property of data table columns, we could also make it a part of the data source specific “meta” property for now.

++ yeah the plan was to do something along these lines

Reply:
yes, this is currently paused a bit due to some other things getting priority but `meta.params` should present the format paramaters and we plan to provide utility functions to work with all the meta information easily, something like this:

```
const fieldFormatter = getFieldFormatter(table.columns[1]);
const aggConfig = getAggConfig(table.columns[1]);
```

Reply:
Great, thanks